hvplot/ui.py

```diff
@@ -95,66 +197,133 @@ class hvPlotExplorer(Viewer):
 
     groupby = param.ListSelector(default=[])
 
-    style = param.ClassSelector(class_=Style)
-
     axes = param.ClassSelector(class_=Axes)
 
+    colormapping = param.ClassSelector(class_=Colormapping)
+
     labels = param.ClassSelector(class_=Labels)
 
+    geo = param.ClassSelector(class_=Geo)
+
     operations = param.ClassSelector(class_=Operations)
 
+    style = param.ClassSelector(class_=Style)
+
+    def __new__(cls, data, **params):
+        if is_geodataframe(data):
+            cls = hvGeomExplorer
+        elif is_xarray(data):
+            cls = hvGridExplorer
+        else:
+            cls = hvDataFrameExplorer
+        return cls(data, **params)
```

Comment:
This code answers my immediate reaction to "Separate DataFrame, Geometry and Gridded explorers while keeping hvPlotExplorer base class", i.e., to ask for a factory that returns something of the correct type.  One step ahead of me!

Still, if hvGeomExplorer is not a subclass of hvDataFrameExplorer, it seems problematic to switch between those two here.  Conceptually, can't any dataframe become a geo dataframe (conceptually) by tacking on a geometry column?  And so shouldn't the capabilities of hvGeomExplorer include _all_ the capabilities of hvDataFrameExplorer, plus a few geo-related ones? I can't tell much from the implementations of these two classes about what their future is, but it seems at first glance that people could add functionality to hvDataFrameExplorer that would then not be available in hvGeomExplorer, which doesn't seem right to me.